### ISSUE TITLE
Make DirectionalLight3D point downwards in its default rotation

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot-proposals#2220]
### Describe the project you are working on

The Godot editor :slightly_smiling_face: 

### Describe the problem or limitation you are having in your project

The default DirectionalLight3D rotation doesn't realistically represent a sun position, since it will light up only walls without lighting up floors. This can confuse beginners.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Make DirectionalLight3D point straight down by default. [While this doesn't represent a realistic rotation in most of the world](https://www.reddit.com/r/pics/comments/64tdgx/when_the_sun_is_directly_overhead_in_hawaii_it/), it will correctly light up a scene unlike a wall-facing DirectionalLight3D.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

There are two ways to implement this proposal:

- Override the default rotation property. I think it's a hacky way to do it, as no other nodes are currently doing this as far as I know.
- Change the shader code and gizmo code. This is quite a lot of work, but it's a cleaner approach compared to the one above. 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

This is about improving the user experience for beginners.

**Comment by aaronfranke** [comment]
With the light as it currently is, the light points in the direction of its `-transform.basis.z` vector, which for an un-rotated light is `Vector3.FORWARD`, which makes sense to me. This also makes it convenient to use with `look_at` and similar.

The current behavior makes sense when dealing with Euler angles in YXZ format, because the Y rotation always represents the azimuth (though it's counter-clockwise instead of the azimuth's typical definition of clockwise), the X rotation always represents the angular altitude (though it's negative, as -90 is directly overhead), and Z always does nothing (it just spins around). However, if the default direction with no rotation had the sun pointing down, then the situation would be that the Y rotation *usually* represents the azimuth (still counter-clockwise), the X rotation *usually* represents the angular altitude (with 0 as overhead and 90 as horizontal), but both of those assumptions break down if the Z rotation is anything but zero, and if X is negative instead of positive then Y would define the azimuth from north instead of south. I don't like this situation I described because it is much easier for people to get into a configuration where the rotation numbers don't correspond with the sun's angular position.

Additionally, people usually will not want a light to be pointing *directly* down, as it looks [strange and not natural](https://twitter.com/tweetsauce/status/542708911699017729/photo/2). With a sun directly overhead by default, grabbing either the X or Z handles and rotating them in either direction would bring the sun down, but doing anything but rotating the X into the positives will give confusing behavior as I described above. Either way, since needing an off-axis rotation is so common for this node, I think it makes sense to start DirectionalLight nodes with an off-axis rotation.

All things considered, I think option 1 is a much better idea, creating a new DirectionalLight node should give it a default rotation.
